swap/tests/concurrent_bobs_after_xmr_lock_proof_sent.rs

```diff
@@ -0,0 +1,62 @@
+pub mod harness;
+
+use harness::bob_run_until::is_xmr_locked;
+use harness::SlowCancelConfig;
+use swap::protocol::alice::AliceState;
+use swap::protocol::bob::BobState;
+use swap::protocol::{alice, bob};
+
+#[tokio::test]
+async fn concurrent_bobs_after_xmr_lock_proof_sent() {
+    harness::setup_test(SlowCancelConfig, |mut ctx| async move {
+        let (bob_swap_1, bob_join_handle_1) = ctx.bob_swap().await;
+        let bob_bitcoin_wallet = bob_swap_1.bitcoin_wallet.clone();
+        let bob_swap_id_1 = bob_swap_1.swap_id;
+        let bob_swap_1 = tokio::spawn(bob::run_until(bob_swap_1, is_xmr_locked));
+
+        let alice_swap_1 = ctx.alice_next_swap().await;
+        let alice_swap_1 = tokio::spawn(alice::run(alice_swap_1));
+
+        let bob_state_1 = bob_swap_1.await??;
+        assert!(matches!(bob_state_1, BobState::XmrLocked { .. }));
+
+        // make sure Bob's swap one event loop is gone
+        bob_join_handle_1.abort();
+        // sync wallet to ensure that it can be re-used for the second swap, otherwise
+        // the lock tx will fail
+        bob_bitcoin_wallet.sync().await?;
```

Comment:
Can we do this implicitly as part of `ctx.bob_swap()`?

That would save as a couple of lines in this test and make it easier to understand.